server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -329,6 +356,70 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (getAdaptiveSelectionStats() != null) {
             getAdaptiveSelectionStats().toXContent(builder, params);
         }
+        if (getShardsStats() != null) {
+            builder.startObject("shards");
+            {
```

Comment:
Do you need this new scope?

Reply:
Technically no, but it has become a common practice within the ES team to do that for these builder objects to make it easier to understand what they're building.